### PR TITLE
more or less correct websocket intereface.  to do - provide simple ja…

### PR DESCRIPTION
**Pull request by Roskowski101**
…vascript example of a client

### CODE REVIEW

File: source/includes/_api_poll.md [hunk 1]
```diff
@@ -3,27 +3,30 @@
...
+
+Websocket poll has a somewhat more flexible model.  The websocket connection implements the session logic. JSON formatted messages from the client modify the state of the session.  As new events are added to the session, their initial state is pushed from the server, and subsequent messages are pushed on any event change. There are two websocket interfaces ws_poll (deprecated) and ws_poll2.  The older ws_poll interface mirrors the statefull symantics of the http interface - all events specified in a subscription will be update in the initial response. The ws_poll2 interface can spread the response over many messages. This results in ws_poll2 being much more responsive on initial start up.  ws_poll2 also allows deletion of events from the session, and will maintain the subscription even if some event is not available.
```
Comment (thartley) [inline source/includes/_api_poll.md]: One of my tasks for the camera settings section was to get rid of all the verbiage around camera settings (per Dean), as he said it was too verbose for the purposes of these docs. I can imagine the same may apply here, so just making note of that here in case we want to revisit this in the next revision.

File: source/includes/_api_poll.md [hunk 2]
```diff
@@ -357,4 +397,244 @@ HTTP Status Code    | Data Type
...
+{
+    "keys": {
+        "<dname key>": {
+            "op": "<operation>"
+        }
+    }
+}
+```
+
+The drop entity indicates a subscribed resource should be removed from the session.
```
Comment (EE-GSlomin) [inline source/includes/_api_poll.md]: This table does not render correctly.

File: source/includes/_api_poll.md [hunk 3]
```diff
@@ -357,4 +397,244 @@ HTTP Status Code    | Data Type
...
+id 		| esn(string) 		| esn hint for infrastructure, pick any cameraid
+
+### HTTP Status Codes
+
+HTTP Status Code    | Notes
+------------------- | ----------- 
+101 | (switch to websocket - invisible)
+404 | Not Found (esn is invalid)
+502 | Resource not available 
+503 | Resource not available (internal only)
```
Comment (thartley) [inline source/includes/_api_poll.md]: internal only, as in public API users shouldn't know about it?

File: source/includes/_api_poll.md [hunk 4]
```diff
@@ -357,4 +397,244 @@ HTTP Status Code    | Data Type
 401	| Unauthorized due to invalid session cookie
 403	| Forbidden due to the user missing the necessary privileges
 
+## WebSocket Interface
```
Comment (thartley) [inline source/includes/_api_poll.md]: I believe the decision was made to only document the v2 websocket API.

File: source/includes/_api_poll.md [hunk 5]
```diff
@@ -357,4 +397,244 @@ HTTP Status Code    | Data Type
...
+
+HTTP Status Code    | Notes
+------------------- | ----------- 
+101 | (switch to websocket - invisible)
+404 | Not Found (esn is invalid)
+502 | Resource not available 
+503 | Resource not available (internal only)
+
+### Overview
+The Websocket2 interface is an API compatible redesign of websocket poll.  The primary change is "camera" entities are moved to independent interactions over the websocket, and will respond on their own (so all cameras may not respond at the same time) and can fail independently.  Initial start up is significantly faster in most cases (as all resolution of cameras is done in parallel) and the session remains intact if there is an issue with an individual camera or request.
```
Comment (thartley) [inline source/includes/_api_poll.md]: once we remove the v1 docs above, reword this to not refer to that v1 websocket poll.

File: source/includes/_api_poll.md [hunk 6]
```diff
@@ -357,4 +397,244 @@ HTTP Status Code    | Data Type
...
+    "drop": {
+        "keys": [ "array of key names to drop" ],
+        "cameras": [ "array of camera esns to drop" ]
+    }
+}
+```
+
+
+
+### Websocket2 Status Codes
```
Comment (EE-GSlomin) [inline source/includes/_api_poll.md]: How do these two sets of error codes differ? I'm assuming one is an http status code while the other is returned within the JSON, but which is which?

File: source/includes/_api_poll.md [hunk 7]
```diff
@@ -77,63 +80,100 @@ IF "Invalid" (bit 16)==1 THEN no status change (use whatever status bits were se
...
-AEDN      | Device alert notification event
-MRBX      | Motion Box event
-MRSZ      | Motion size reports
-ROMS      | Region of interest motion start
-ROME      | Region of interest motion end
-ALMS      | Alert Motion Start
-ALME      | Alert Motion End
-ALRS      | Alert Region Of Interest Start
-ALRE      | Alert Region Of Interest End
+fourcc | name
```
Comment (thartley) [inline source/includes/_api_poll.md]: should we add a short description for each of these?

**Comment by klaushofrichter** [comment]
@thartley , @Roskowski101  - I am not sure why this now shows in my list of open PRs, but it's probably safe to close this one?

**Comment by klaushofrichter** [comment]
@Roskowski101 , @thartley  - can we close this PR?

**Comment by thartley** [comment]
> @Roskowski101 , @thartley - can we close this PR?

@klaushofrichter considering how old this PR is, my opinion is that yes, we can close it.   However, since it's not my PR, I can't make that call.
